zulipterminal/helper.py

```diff
@@ -393,22 +394,51 @@ def powerset(iterable: Iterable[Any],
     powerset = chain.from_iterable(combinations(s, r) for r in range(len(s)+1))
     return list(map(map_func, list(powerset)))
 
-def open_media(url: str):
-    img_name = url.split("/")[-1]
-    img_dir_path = os.path.join(tempfile.gettempdir(), "ZulipTerminal_media")
-    img_path = os.path.join(img_dir_path, img_name)
+@contextmanager
+def suppress_output() -> Iterator[Any]:
+    """
+    Context manager to redirect stdout and stderr to /dev/null.
+    Adapted from https://stackoverflow.com/a/2323563
+    """
+    out = os.dup(1)
+    err = os.dup(2)
+    os.close(1)
+    os.close(2)
+    os.open(os.devnull, os.O_RDWR)
     try:
-        os.mkdir(img_dir_path)
-    except FileExistsError:
-        pass
-    with requests.get(url, stream=True) as r:
-        if r.status_code == 200:
-            with open(img_path, 'wb') as f:
-                for chunk in r.iter_content(chunk_size=1024):
-                    if chunk:  # filter out keep-alive new chunks
-                        f.write(chunk)
-    r.close()
-    imageViewerFromCommandLine = {'linux': 'xdg-open',
-                                    'win32': 'explorer',
-                                    'darwin': 'open'}[sys.platform]
-    subprocess.run([imageViewerFromCommandLine, img_path])
+        yield
+    finally:
+        os.dup2(out, 1)
+        os.dup2(err, 2)
+
+def open_media(controller, url: str):
+    # Uploaded media
+    if 'user_uploads' in url:
+        # Tries to download media and open in default viewer
+        # If can't, opens in browser
+        img_name = url.split("/")[-1]
+        img_dir_path = os.path.join(tempfile.gettempdir(), "ZulipTerminal_media")
+        img_path = os.path.join(img_dir_path, img_name)
+        try:
+            os.mkdir(img_dir_path)
+        except FileExistsError:
+            pass
+        with requests.get(url, stream=True) as r:
```

Comment:
How about:
```python
        with requests.get(url,
                          auth=requests.auth.HTTPBasicAuth(client.email,
                                                          client.api_key),
                          stream=True) as r:
```

where client is controller.client.

Reply:
That works. Thanks a lot, @amanagr!